Bug: kor comma list results in crash

**Describe the bug**
When setting 2 different resources, kor crashes. 



**To Reproduce**
```
kor ingress,role 
```


```

panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x20 pc=0x212e1a3]

goroutine 1 [running]:
github.com/yonahd/kor/pkg/kor.retrieveUsedRoles({0x2776718?, 0xc0005031e0?}, {0xc000238250, 0xb}, 0x0)
	github.com/yonahd/kor/pkg/kor/roles.go:26 +0x263
github.com/yonahd/kor/pkg/kor.processNamespaceRoles({0x2776718, 0xc0005031e0}, {0xc000238250, 0xb}, 0x30?)
	github.com/yonahd/kor/pkg/kor/roles.go:63 +0x3f
github.com/yonahd/kor/pkg/kor.getUnusedRoles({0x2776718?, 0xc0005031e0?}, {0xc000238250, 0xb}, 0x235a800?)
	github.com/yonahd/kor/pkg/kor/all.go:77 +0x51
github.com/yonahd/kor/pkg/kor.retrieveNamespaceDiffs({0x2776718, 0xc0005031e0}, {0xc000238250, 0xb}, {0xc000018920?, 0x2, 0x0?})
	github.com/yonahd/kor/pkg/kor/multi.go:37 +0x1f0
github.com/yonahd/kor/pkg/kor.GetUnusedMulti({{0x0?, 0x0?}, {0x0?, 0x0?}}, {0x0?, 0x0?}, {0x7ff7bfeff2bc, 0xc}, {0x0, 0x0, ...})
	github.com/yonahd/kor/pkg/kor/multi.go:70 +0x14c
github.com/yonahd/kor/cmd/kor.glob..func10(0xc000240f00?, {0xc000277130, 0x4?, 0x250151d?})
	github.com/yonahd/kor/cmd/kor/root.go:31 +0x1e5
github.com/spf13/cobra.(*Command).execute(0x33bfb40, {0xc000040050, 0x1, 0x1})
	github.com/spf13/cobra@v1.8.0/command.go:987 +0xaa3
github.com/spf13/cobra.(*Command).ExecuteC(0x33bfb40)
	github.com/spf13/cobra@v1.8.0/command.go:1115 +0x3ff
github.com/spf13/cobra.(*Command).Execute(...)
	github.com/spf13/cobra@v1.8.0/command.go:1039
github.com/yonahd/kor/cmd/kor.Execute()
	github.com/yonahd/kor/cmd/kor/root.go:64 +0x2d1
main.main()
	github.com/yonahd/kor/main.go:6 +0xf
```

--- a/cmd/kor/root.go
+++ b/cmd/kor/root.go
@@ -21,14 +21,12 @@ var rootCmd = &cobra.Command{
 
 		// Cheks whether the string contains a comma, indicating that it represents a list of resources
 		if strings.ContainsRune(resourceNames, 44) {
-			if outputFormat == "json" || outputFormat == "yaml" {
-				if response, err := kor.GetUnusedMultiStructured(includeExcludeLists, kubeconfig, outputFormat, resourceNames); err != nil {
-					fmt.Println(err)
-				} else {
-					fmt.Println(response)
-				}
+			clientset := kor.GetKubeClient(kubeconfig)
+
+			if response, err := kor.GetUnusedMulti(includeExcludeLists, resourceNames, filterOptions, clientset, outputFormat, opts); err != nil {
+				fmt.Println(err)
 			} else {
-				kor.GetUnusedMulti(includeExcludeLists, kubeconfig, resourceNames, opts)
+				fmt.Println(response)
 			}
 		} else {
 			fmt.Printf("Subcommand %q was not found, try using 'kor --help' for available subcommands", args[0])

--- a/pkg/kor/multi.go
+++ b/pkg/kor/multi.go
@@ -8,45 +8,44 @@ import (
 	"strings"
 
 	"k8s.io/client-go/kubernetes"
-	"sigs.k8s.io/yaml"
 )
 
-func retrieveNamespaceDiffs(clientset kubernetes.Interface, namespace string, resourceList []string) []ResourceDiff {
+func retrieveNamespaceDiffs(clientset kubernetes.Interface, namespace string, resourceList []string, filterOpts *FilterOptions) []ResourceDiff {
 	var allDiffs []ResourceDiff
 	for _, resource := range resourceList {
 		switch resource {
 		case "cm", "configmap", "configmaps":
-			namespaceCMDiff := getUnusedCMs(clientset, namespace, nil)
+			namespaceCMDiff := getUnusedCMs(clientset, namespace, filterOpts)
 			allDiffs = append(allDiffs, namespaceCMDiff)
 		case "svc", "service", "services":
 			namespaceSVCDiff := getUnusedSVCs(clientset, namespace)
 			allDiffs = append(allDiffs, namespaceSVCDiff)
 		case "scrt", "secret", "secrets":
-			namespaceSecretDiff := getUnusedSecrets(clientset, namespace, nil)
+			namespaceSecretDiff := getUnusedSecrets(clientset, namespace, filterOpts)
 			allDiffs = append(allDiffs, namespaceSecretDiff)
 		case "sa", "serviceaccount", "serviceaccounts":
 			namespaceSADiff := getUnusedServiceAccounts(clientset, namespace)
 			allDiffs = append(allDiffs, namespaceSADiff)
 		case "deploy", "deployment", "deployments":
-			namespaceDeploymentDiff := getUnusedDeployments(clientset, namespace, nil)
+			namespaceDeploymentDiff := getUnusedDeployments(clientset, namespace, filterOpts)
 			allDiffs = append(allDiffs, namespaceDeploymentDiff)
 		case "sts", "statefulset", "statefulsets":
-			namespaceStatefulsetDiff := getUnusedStatefulSets(clientset, namespace, nil)
+			namespaceStatefulsetDiff := getUnusedStatefulSets(clientset, namespace, filterOpts)
 			allDiffs = append(allDiffs, namespaceStatefulsetDiff)
 		case "role", "roles":
-			namespaceRoleDiff := getUnusedRoles(clientset, namespace, nil)
+			namespaceRoleDiff := getUnusedRoles(clientset, namespace, filterOpts)
 			allDiffs = append(allDiffs, namespaceRoleDiff)
 		case "hpa", "horizontalpodautoscaler", "horizontalpodautoscalers":
-			namespaceHpaDiff := getUnusedHpas(clientset, namespace, nil)
+			namespaceHpaDiff := getUnusedHpas(clientset, namespace, filterOpts)
 			allDiffs = append(allDiffs, namespaceHpaDiff)
 		case "pvc", "persistentvolumeclaim", "persistentvolumeclaims":
-			namespacePvcDiff := getUnusedPvcs(clientset, namespace, nil)
+			namespacePvcDiff := getUnusedPvcs(clientset, namespace, filterOpts)
 			allDiffs = append(allDiffs, namespacePvcDiff)
 		case "ing", "ingress", "ingresses":
-			namespaceIngressDiff := getUnusedIngresses(clientset, namespace, nil)
+			namespaceIngressDiff := getUnusedIngresses(clientset, namespace, filterOpts)
 			allDiffs = append(allDiffs, namespaceIngressDiff)
 		case "pdb", "poddisruptionbudget", "poddisruptionbudgets":
-			namespacePdbDiff := getUnusedPdbs(clientset, namespace, nil)
+			namespacePdbDiff := getUnusedPdbs(clientset, namespace, filterOpts)
 			allDiffs = append(allDiffs, namespacePdbDiff)
 		default:
 			fmt.Printf("resource type %q is not supported\n", resource)
@@ -55,50 +54,28 @@ func retrieveNamespaceDiffs(clientset kubernetes.Interface, namespace string, re
 	return allDiffs
 }
 
-func GetUnusedMulti(includeExcludeLists IncludeExcludeLists, kubeconfig, resourceNames string, opts Opts) {
-	var clientset kubernetes.Interface
-	var namespaces []string
-
+func GetUnusedMulti(includeExcludeLists IncludeExcludeLists, resourceNames string, filterOpts *FilterOptions, clientset kubernetes.Interface, outputFormat string, opts Opts) (string, error) {
 	var outputBuffer bytes.Buffer
-
-	clientset = GetKubeClient(kubeconfig)
-
 	resourceList := strings.Split(resourceNames, ",")
-	namespaces = SetNamespaceList(includeExcludeLists, clientset)
+	namespaces := SetNamespaceList(includeExcludeLists, clientset)
+	response := make(map[string]map[string][]string)
+	var err error
 
 	for _, namespace := range namespaces {
-		allDiffs := retrieveNamespaceDiffs(clientset, namespace, resourceList)
-		output := FormatOutputAll(namespace, allDiffs)
+		allDiffs := retrieveNamespaceDiffs(clientset, namespace, resourceList, filterOpts)
 
+		if opts.DeleteFlag {
+			for _, diff := range allDiffs {
+				if diff.diff, err = DeleteResource(diff.diff, clientset, namespace, diff.resourceType, opts.NoInteractive); err != nil {
+					fmt.Fprintf(os.Stderr, "Failed to delete %s %s in namespace %s: %v\n", diff.resourceType, diff.diff, namespace, err)
+				}
+			}
+
+		}
+		output := FormatOutputAll(namespace, allDiffs)
 		outputBuffer.WriteString(output)
 		outputBuffer.WriteString("\n")
-	}
 
-	if opts.WebhookURL != "" || opts.Channel != "" && opts.Token != "" {
-		if err := SendToSlack(SlackMessage{}, opts, outputBuffer.String()); err != nil {
-			fmt.Fprintf(os.Stderr, "Failed to send message to slack: %v\n", err)
-			os.Exit(1)
-		}
-	} else {
-		fmt.Println(outputBuffer.String())
-	}
-}
-
-func GetUnusedMultiStructured(includeExcludeLists IncludeExcludeLists, kubeconfig, outputFormat, resourceNames string) (string, error) {
-	var clientset kubernetes.Interface
-	var namespaces []string
-
-	clientset = GetKubeClient(kubeconfig)
-
-	resourceList := strings.Split(resourceNames, ",")
-	namespaces = SetNamespaceList(includeExcludeLists, clientset)
-
-	// Create the JSON response object
-	response := make(map[string]map[string][]string)
-
-	for _, namespace := range namespaces {
-		allDiffs := retrieveNamespaceDiffs(clientset, namespace, resourceList)
-		// Store the unused resources for each resource type in the JSON response
 		resourceMap := make(map[string][]string)
 		for _, diff := range allDiffs {
 			resourceMap[diff.resourceType] = diff.diff
@@ -106,19 +83,15 @@ func GetUnusedMultiStructured(includeExcludeLists IncludeExcludeLists, kubeconfi
 		response[namespace] = resourceMap
 	}
 
-	// Convert the response object to JSON
 	jsonResponse, err := json.MarshalIndent(response, "", "  ")
 	if err != nil {
 		return "", err
 	}
 
-	if outputFormat == "yaml" {
-		yamlResponse, err := yaml.JSONToYAML(jsonResponse)
-		if err != nil {
-			fmt.Printf("err: %v\n", err)
-		}
-		return string(yamlResponse), nil
-	} else {
-		return string(jsonResponse), nil
+	unusedMulti, err := unusedResourceFormatter(outputFormat, outputBuffer, opts, jsonResponse)
+	if err != nil {
+		fmt.Printf("err: %v\n", err)
 	}
+
+	return unusedMulti, nil
 }

